src/utils/storage.js

```diff
@@ -0,0 +1,38 @@
+function setItem(key, value) {
+  localStorage.setItem(key, JSON.stringify(value));
+  return value;
+}
+
+function getTodos(key) {
+  return JSON.parse(localStorage.getItem(key)) || [];
+}
+
+function addNewTodo(key, todo) {
+  const todos = getTodos(key);
+
+  return setItem(key, [...todos, todo]);
+}
+
+function deleteTodo(key, todo) {
+  const todos = getTodos(key);
+  const updatedItems = todos.filter((_todo) => todo.id !== _todo.id);
+  return setItem(key, updatedItems);
+}
+
+function updateAchievedTodo(key, todo) {
+  const todos = getTodos(key);
+  const updatedItems = todos.map((_todo) => {
+    return _todo.id === todo.id ? { ..._todo, done: true } : _todo;
```

Comment:
I would suggest updating the `done` value by using the opposite value instead of only setting it to `true`. In other words, you can toggle the value of `done`. The issue with this approach of only setting the `done` value to `true`, is once the user clicks the checkbox of the to-do item, the item will be marked as complete but there will be no option for the user to unmark the item.

```suggestion
    return _todo.id === todo.id ? { ..._todo, done: !_todo.done } : _todo;
```